tests/test_main.py

```diff
@@ -8,7 +8,9 @@
 
 
 def test_dexit() -> None:
-    with pytest.raises(SystemExit, match=re.escape("\x1b[31mERROR:\x1b[0m message")):
+    with pytest.raises(
+        SystemExit, match=re.escape("\x1b[38;2;170;0;0mERROR:\x1b[0m message")
+    ):
```

Comment:
There's an issue here: while dahlia v2 defaulted to 16-bit colors, dahlia v3 tries to detect the color depth supported by the terminal emulator. So whether the tests pass now depends on what emulator one's using (so e.g. this will fail in Terminal.app, but not in Ghostty or Alacritty). I think we can drop the `re.escape` and just match for any ANSI code:
```suggestion
    with pytest.raises(
        SystemExit, match=r"\x1b\[.*mERROR:\x1b\[0m message"
    ):
```
Same applies for the other tests